paddle/phi/kernels/fusion/onednn/fused_matmul_kernel.cc

```diff
@@ -166,6 +166,16 @@ class FusedMatmulOneDNNHandler
     }
 
     if (residual_data) {
+      // fill 1 in the front of adesc, to make residual ndims to be same as dst
+      // dims
+      int dst_size = out_ddims.size();
+      int origin_size = residual_data->mem_desc().get_ndims();
+      dnnl::memory::dims expanded_dims = residual_data->mem_desc().get_dims();
+      if (origin_size < dst_size) {
```

Comment:
Hi @zhanglirong1999 , currently this fix is mainly targeting to the reported issue, where the dimension of residual_data is [H] while dst is [1, 1, H] (or [1, H] and [1, 1, H] also works), the similar dimension is just right on the last. However, if the dst is [1, H, 1] or [H, 1, 1] or more shape for higher dimensions. This fix may not be about to solve such situations. 

Usually I assume such kind of incompatibility of dimensions happens when residual_data is scale, so maybe and one more condition could extend the range of this fix's application?
```suggestion
      if (origin_size < dst_size) {
        if (origin_size == 1) {
          # compare which dimension of dst is same as residual_size
          # expand the dims based on result
          # reshape the md
        } else {
        expanded_dims.insert(expanded_dims.begin(), dst_size - origin_size, 1);
        residual_data->mem_desc().reshape(expanded_dims);
        }}
```

Reply:
yeah, I think it's a good consideration, but in this way we also need to consider more multi-dimensional and multi-dimensional matching. It seems that OneDNN only considers one dimension to expand, treating them as scale~ 
https://github.com/oneapi-src/oneDNN/blob/master/src/graph/backend/dnnl/common.cpp#L229

Reply:
Yea you're right~ Then we should just keep the same as what oneDNN does~